vars/stepResult.groovy

```diff
@@ -150,5 +150,7 @@ def call(Map config= [:]) {
   }
   param['status'] = result
 
+  writeFile(file: stageStatusFilename(), text: result)
```

Comment:
No, what is written to Jenkins job status by default is the Jenkins stage status, this is the status reported to the github step.
It would take quite a bit of changes to the Jenkins job status procedure that would need to be rolled out to all the Jenkinsfiles to make it both expandable and usable for both purposes, so for now they need to be kept separate.